gcloud/bigquery/job.py

```diff
@@ -0,0 +1,702 @@
+        """
+        if not isinstance(value, six.integer_types):
+            raise ValueError("Pass a boolean")
+        self._configuration._skip_leading_rows = value
+
+    @skip_leading_rows.deleter
+    def skip_leading_rows(self):
+        """Delete skip_leading_rows."""
+        del self._configuration._skip_leading_rows
+
+    @property
+    def source_format(self):
+        """Format of source data files.
+
+        :rtype: string, or ``NoneType``
+        :returns: The value as set by the user, or None (the default).
+        """
+        return self._configuration._source_format
+
+    @source_format.setter
+    def source_format(self, value):
+        """Update source_format.
+
+        :type value: string
+        :param value: new source_format: one of "CSV", "DATASTORE_BACKUP",
+                      or "NEWLINE_DELIMITED_JSON"
+
+        :raises: ValueError for invalid values.
+        """
+        if value not in ('CSV', 'DATASTORE_BACKUP', 'NEWLINE_DELIMITED_JSON'):
+            raise ValueError(
+                "Pass 'CSV', 'DATASTORE_BACKUP' or 'NEWLINE_DELIMITED_JSON'")
+        self._configuration._source_format = value
+
+    @source_format.deleter
+    def source_format(self):
+        """Delete source_format."""
+        del self._configuration._source_format
+
+    @property
+    def write_disposition(self):
+        """Allow rows with missing trailing commas for optional fields.
+
+        :rtype: boolean, or ``NoneType``
+        :returns: The value as set by the user, or None (the default).
+        """
+        return self._configuration._write_disposition
+
+    @write_disposition.setter
+    def write_disposition(self, value):
+        """Update write_disposition.
+
+        :type value: string
+        :param value: new write_disposition: one of "WRITE_APPEND",
+                      "WRITE_TRUNCATE", or "WRITE_EMPTY"
+
+        :raises: ValueError for invalid value types.
+        """
+        if value not in ('WRITE_APPEND', 'WRITE_TRUNCATE', 'WRITE_EMPTY'):
+            raise ValueError(
+                "Pass 'WRITE_APPEND', 'WRITE_TRUNCATE' or 'WRITE_EMPTY'")
+        self._configuration._write_disposition = value
+
+    @write_disposition.deleter
+    def write_disposition(self):
+        """Delete write_disposition."""
+        del self._configuration._write_disposition
+
+    def _require_client(self, client):
+        """Check client or verify over-ride.
+
+        :type client: :class:`gcloud.bigquery.client.Client` or ``NoneType``
+        :param client: the client to use.  If not passed, falls back to the
+                       ``client`` stored on the current dataset.
+
+        :rtype: :class:`gcloud.bigquery.client.Client`
+        :returns: The client passed in or the currently bound client.
+        """
+        if client is None:
+            client = self._client
+        return client
+
+    def _populate_config_resource(self, configuration):
```

Comment:
I think the property implementations, along with this method, will end up on a `_LoadJobBase` class which shares the implementation across the three(?) load job types.